src/classes/UTIL_OrgTelemetry_SObject_BATCH.cls

```diff
@@ -74,10 +88,15 @@ public class UTIL_OrgTelemetry_SObject_BATCH implements Database.Batchable<SObje
      */
     public Database.QueryLocator start(Database.BatchableContext context) {
         switch on (jobMode) {
-            when RECURRING_DONATIONS {
+            when RECURRING_DONATIONS_OPPS {
                 String soql = 'SELECT Id, npe03__Recurring_Donation__r.npe03__Open_Ended_Status__c FROM Opportunity';
                 return Database.getQueryLocator(soql);
             }
+
+            when RECURRING_DONATIONS {
+                String soql = 'SELECT npe03__Amount__c, (SELECT Amount FROM npe03__Donations__r) FROM npe03__Recurring_Donation__c';
```

Comment:
Using a sub-query in a queryLocator, especially in a large org, can cause query timeout errors. The recommended best practice for this is to query the main object -- `npe03__Recurring_Donation__c` -- in the QueryLocator and then execute a second query in the `execute()` method to retrieve the related records.

The one concern I have is that we could hit a governor limit in the execute() method if there are more than 50K related Opportunities for the collection of 2000 Recurring Donation Id's. To that end, I recommend changing the batch size of the RecurringDonations job to 200 instead of 2000. That would allow for up to 250 Opportunities per RD, assuming every RD in the batch had that many. Plus generally there are significantly less RD's in an Org than Opps, so a batch size of 200 should still finish pretty fast.